droidlet/dialog/chat_schema.md

```diff
@@ -0,0 +1,71 @@
+## Description ##
+This is the source of truth for the most updated version of the schema to be followed when sending chats from the agent the the user (and perhaps eventually from the agent to other agents).  When updating this document please increment the version number.  Older versions of the schema can be found in git history.
+
+Note that chats sent from the user to the agent should always be in plaintext, as all messages should be parsed by the NSP and interpreted.  I.e. special pathways and conditionals on agent input are discouraged.  It is possible to help the NSP by including special tokens that are reserved for specific use cases, eg. clarification response.
+
+### Content Types ###
+ - `chat_string`: Only text to display in the chat window with no response assumed.
+    - Example: "I finished building this!"
+ - `chat_and_media`: Text and media to display in the chat window, with no response assumed.
+    - Example: "I finished building this!" (with accompanying photo)
+ - `chat_and_text_options`: Text to display in the chat window along with a list of possible response options.
+    - Example: "Are these the Droidlets you are looking for?" ('no', 'no')
+ - `chat_and_media_options`: Text to display in the chat window along with a list of clickable media options.
+    - Example: "Which of these is the 'blue sphere' you were referring to?" (pictures of blue spheres)
+ - `chat_and_media_and_text_options`: Text and media to display in the chat window alond with a list of possible response options.
+    - Example: "Is this funny?" <img src='https://external-preview.redd.it/hWh_8TpqrT6zAwpzHJ_m9Rx3iHjc_yI4zSI6aazMFTc.jpg?auto=webp&s=6d8006ac3edca5bad98dd7b5b9a4a8d5554eaff0' width="200"> ('yes', 'no')
+
+### Schema ###
+
+<pre>
+{
+	"title": "Droidlet Chat",
+	"description": "A single chat sent from a Droidlet agent to the user",
+	"version": 1,
+	"type": "object",
+
+	"properties": {
+		"speaker_id": {
+			"description": "The memid of the speaker",
```

Comment:
isn't the speaker going to be the agent here ?

Reply:
Yes, this field may be superfluous now.  I included it for two reasons.  One is that speaker memid, chat text, and timestamp are the three ChatNode arguments, so having the memid accessible could make for convenient retrieval of the memory node.  Though we could presumably get it from the agent at pretty much any time anyway.  The second reason is looking forward to a situation in which more speakers are around than just the agent and the user.  But it's understandable if you want to plan for this use case when it comes around.

Let me know if this is enough justification to keep it or if you'd prefer to drop it for now.

Reply:
I replaced this with the memid of the chat itself.